RustLeanModels/Basic.lean

```diff
@@ -13,19 +12,23 @@ abbrev Str := List Char
 
 --LOGIC
 
-lemma exists_minima_prop_nat {p: Nat → Prop} : (∃ x , p x) → ∃ m, (p m) ∧ (∀ n, p n → m ≤ n) :=by
+theorem exists_minima_prop_nat {p: Nat → Prop} : (∃ x , p x) → ∃ m, (p m) ∧ (∀ n, p n → m ≤ n) :=by
   intro gn; obtain ⟨n, gn⟩ := gn
-  induction n using Nat.strong_induction_on
+  induction n using Nat.strongInductionOn
   rename_i n ind
   by_cases (∀ m < n, ¬ p m ); rename_i gm
   · exists n ; simp only [gn, true_and]; intro x gx ;
-    by_contra; rename_i gc; simp only [not_le] at gc;
-    have gm := gm x gc; contradiction
-  · rename_i gm ; simp only [not_forall, not_not, exists_prop] at gm; obtain ⟨m,gm⟩:=gm
-    have ind:= ind m gm.left gm.right ; obtain⟨x,gx⟩:= ind
-    use x;
-
-lemma imp_eq_not_or: (p → q) ↔  (¬ p ∨ q) :=by
+    have gm := gm x
+    simp only [gx, not_true_eq_false, imp_false, Nat.not_lt] at gm
+    exact gm
+  · rename_i gm;
+    simp only [Classical.not_forall, not_imp, Classical.not_not] at gm
+    obtain ⟨m,gm⟩:=gm
+    simp only [exists_prop, true_implies] at gm
+    have ind := ind m gm.left gm.right; obtain ⟨ x, gx ⟩ := ind
```

Comment:
Same trick here. Line 28 can be simplified to
```
      have ⟨ x, gx ⟩ := ind m gm.left gm.right
```

Reply:
Done.